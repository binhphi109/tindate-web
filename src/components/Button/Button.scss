.button {
  flex: 1;
  display: flex;
  justify-content: center;

  i {
    width: 24px;
  }
}
