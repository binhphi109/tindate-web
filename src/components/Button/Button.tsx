import React, { MouseEventHandler } from "react";
import { IconButton } from "@material-ui/core";
import clsx from "clsx";
import "./Button.scss";

type Props = {
  className?: string;
  icon: string;
  iconColor?: string;
  color?: "inherit" | "primary" | "secondary" | "default";
  onClick?: MouseEventHandler;
};

function Button({ className, icon, iconColor, color, onClick }: Props) {
  return (
    <div className={clsx("button", className)} onClick={onClick}>
      <IconButton size="medium" color={color}>
        <i className={icon} style={{ color: iconColor }} />
      </IconButton>
    </div>
  );
}

Button.defaultProps = {
  className: null,
  iconColor: null,
  color: "default",
  onClick: null,
};

export default Button;
