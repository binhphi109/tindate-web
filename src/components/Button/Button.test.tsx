import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

describe("components/button/Button", () => {
  test("should match snapshot", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button
        className="reactionButton"
        icon="fas fa-times"
        iconColor="#fc575e"
        onClick={onClick}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("should match snapshot without className", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button icon="fas fa-times" iconColor="#fc575e" onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should match snapshot without iconColor ", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button className="reactionButton" icon="fas fa-times" onClick={onClick} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("should match snapshot without onClick ", () => {
    const wrapper = shallow(<Button className="reactionButton" icon="fas fa-times" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should handle onClick", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button
        className="reactionButton"
        icon="fas fa-times"
        iconColor="#fc575e"
        onClick={onClick}
      />
    );

    wrapper.find("div.button").simulate("click");
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
