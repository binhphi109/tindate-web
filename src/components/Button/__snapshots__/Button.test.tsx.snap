// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/button/Button should match snapshot 1`] = `
<div
  className="button reactionButton"
  onClick={[MockFunction]}
>
  <WithStyles(ForwardRef(IconButton))
    color="default"
    size="medium"
  >
    <i
      className="fas fa-times"
      style={
        Object {
          "color": "#fc575e",
        }
      }
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`components/button/Button should match snapshot without className 1`] = `
<div
  className="button"
  onClick={[MockFunction]}
>
  <WithStyles(ForwardRef(IconButton))
    color="default"
    size="medium"
  >
    <i
      className="fas fa-times"
      style={
        Object {
          "color": "#fc575e",
        }
      }
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`components/button/Button should match snapshot without iconColor  1`] = `
<div
  className="button reactionButton"
  onClick={[MockFunction]}
>
  <WithStyles(ForwardRef(IconButton))
    color="default"
    size="medium"
  >
    <i
      className="fas fa-times"
      style={
        Object {
          "color": null,
        }
      }
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`components/button/Button should match snapshot without onClick  1`] = `
<div
  className="button reactionButton"
  onClick={null}
>
  <WithStyles(ForwardRef(IconButton))
    color="default"
    size="medium"
  >
    <i
      className="fas fa-times"
      style={
        Object {
          "color": null,
        }
      }
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;
